tools/tfrecord2idx

```diff
@@ -1,13 +1,15 @@
 #!/usr/bin/env python
 import sys
 import struct
+import tensorflow as tf
+import tensorflow_io as tfio  # noqa:F401
 
 if len(sys.argv) < 3:
     print("Usage: tfrecord2idx <tfrecord filename> <index filename>")
     exit()
 
-f = open(sys.argv[1], 'rb')
-idx = open(sys.argv[2], 'w')
+f = tf.io.gfile.GFile(sys.argv[1], 'rb')
+idx = tf.io.gfile.GFile(sys.argv[2], 'w')
```

Comment:
There is one more thing that I realized after running it on our CI. This change makes the script dependent on DALI. While people may have data in TFRecord they not necessarily need to have TF installed.
I wonder if you can make it options:
- either add an cmd line option to import and use tf appi when user asks for it explicltly
- try importing tf and if it fails use the POSIX API

What do you think?

Reply:
@JanuszL I pushed the changes adopting option #2. In this case, we limit the changes to ``tfrecord2idx`` script, so not affecting its callers.

Reply:
Thank you. Let me run it through the CI.